### PR TITLE
Added Test case for check iam_avoid_root_usage

### PR DESCRIPTION
**Pull request by Gunjan-Katre-Comprinno**
Context
This PR introduces unit tests for the iam_avoid_root_usage check, which validates whether the AWS root account has been used within the past 7 days. Minimizing root account usage is critical to securing AWS environments.

Description
The following test cases are included to cover core logic and edge conditions:

-  Root account used recently (within 7 days) – The check should fail.
-  Root account used long ago (more than 7 days ago) – The check should pass.
-  No valid last access timestamps available – The check should return a skipped status.
-  IAM client error during report generation (e.g., AccessDenied) – The check should return an unknown status.

Credential report data is mocked to simulate the AWS IAM response, ensuring consistent and isolated test behavior. This test suite increases confidence in logic that parses credential reports and determines last usage timestamps for the root account.

Checklist

-  Coverage includes pass, fail, skipped, and error states
-  Uses mocked AWS IAM client and credential report content
-  Follows project conventions for setup and assertions
-  Validates summary messages for clear reporting feedback

License
I confirm that this contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: library/aws/tests/iam/test_iam_avoid_root_usage.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    def test_credential_report_generation_failure(self):
+        self.mock_client.generate_credential_report.side_effect = ClientError(
+            {"Error": {"Code": "AccessDenied", "Message": "Access Denied"}}, "GenerateCredentialReport"
+        )
+
+        report = self.check.execute(self.mock_session)
+
+        assert report.status == CheckStatus.UNKNOWN
+        assert any(r.summary is not None and "IAM API request failed" in r.summary for r in report.resource_ids_status)
```
Comment (aafaq-rashid-comprinno) [inline library/aws/tests/iam/test_iam_avoid_root_usage.py:93]: The test case for `iam_avoid_root_usage` mirrors the structure and rigor of `iam_password_policy_lowercase`, covering recent and old root access, unknown usage, and API failure paths with clear mocks, exception handling, and precise assertions.
